concent_api/middleman/middleman_server.py

```diff
@@ -96,6 +110,25 @@ def _close_middleman(self) -> None:
             )
             raise
 
+    async def _handle_service_connection(self, reader, writer) -> None:
+        self._signing_service_connection_counter += 1
+        if self._signing_service_connection_counter > 1:
```

Comment:
As I understand, you want only one connection at a time to be opened. I think it would be easier if instead of `counter` you use `bool`, ie. `is_signing_service_connection_active`.

Reply:
Done